lib/spi/types.ts

```diff
@@ -60,4 +61,53 @@ export interface ApplicationRepository {
      * The Kubernetes version for the cluster.
      */
     readonly version: KubernetesVersion;
+}
+
+/**
+ * ClusterInfo desribes an EKS Cluster
+ */
+export class ClusterInfo {
+    
+    /**
+     * Attributes
+     */
+    readonly cluster: Cluster;
+    readonly version: KubernetesVersion;
+    readonly nodeGroup?: Nodegroup;
+    readonly autoScalingGroup?: AutoScalingGroup;
+    private readonly provisionedAddOns: Map<string, cdk.Construct>;
+
+    /**
+     * Constructor for ClusterInfo
+     * @param props 
+     */
+    constructor(props: ClusterInfoProps){
```

Comment:
Minor: we could be passing variables to the constructor directly with no ClusterInfoProps for less verbosity. E.g. 

```
constructor(cluster, nodeGroup?: NodeGroup | AutoscalingGroup)
```

Reply:
done.